pkg/cmd/server/api/v1/types.go

```diff
@@ -833,6 +833,21 @@ type LDAPSyncConfig struct {
 	// OpenShift Group names
 	LDAPGroupUIDToOpenShiftGroupNameMapping map[string]string `json:"groupUIDNameMapping"`
 
+	// TolerateMissingMembers determines the behavior of the LDAP sync job when missing user entries are
+	// encountered. If 'true', an LDAP query for users that doesn't find any will be tolerated and an only
+	// and error will be logged. If 'false', the LDAP sync job will fail if a query for users doesn't find
+	// any. The default value is 'false'. Misconfigured LDAP sync jobs with this flag set to 'true' can cause
+	// group membership to be removed, so it is recommended to use this flag with caution.
+	TolerateMissingMembers bool `json:"tolerateMissingMembers"`
+
+	// TolerateOutOfScopeMembers determines the behavior of the LDAP sync job when out-of-scope user entries
+	// are encountered. If 'true', an LDAP query for a user that falls outside of the base DN given for the all
+	// user query will be tolerated and only an error will be logged. If 'false', the LDAP sync job will fail
+	// if a user query would search outside of the base DN specified by the all user query. Misconfigured LDAP
+	// sync jobs with this flag set to 'true' can result in groups missing users, so it is recommended to use
+	// this flag with caution.
+	TolerateOutOfScopeMembers bool `json:"tolerateOutOfScopeMembers"`
```

Comment:
> I will say putting the flags into the query breaks our data model a little, currently the query is a glorified constructor for search requests, which can't contain this behavior.

We need this to make as much sense as possible to admin's.  I expect this ended up top level because that's where I said a user-blacklist should go since logically a user-blacklist would span all schemas.